### PR TITLE
fix(lsp): set tabSize from 'shiftwidth', not 'softtabstop'

### PR DESCRIPTION
**Pull request by tpope**
The use of 'softtabstop' to set tabSize was introduced in 5d5b068, replacing 'tabstop'.  This was a step in the right direction, but if we look past the name tabSize and at the actual purpose of the field, it's the indentation width used when formatting.  (The field should have been called "indentSize", but there's nothing to be done about that now.)  This corresponds to the Vim option 'shiftwidth', not 'softtabstop'.  The latter has the comparatively mundane purpose of controlling what happens when you hit the tab key (and even this is incomplete, as it fails to account for 'smarttab').

See also [the coc.nvim implementation](https://github.com/neoclide/coc.nvim/blob/287c743c9f227fdf0e1db452bbb8ae3c5caffc36/autoload/coc/util.vim#L866-L876), which uses 'shiftwidth'.

### CODE REVIEW

**Comment by mjlbach** [comment]
Thanks! The change itself looks good to me, but do you think we should go ahead and rename the the functions/tests as well. We could do `get_effective_tabsize` (idiomatic with the lsp specification naming) or `get_effective_indentsize`. 

Here are the usages:
```
runtime/doc/lsp.txt|1457 col 1| get_effective_tabstop({bufnr})          *vim.lsp.util.get_effective_tabstop()*
runtime/lua/vim/lsp/util.lua|1902 col 12| function M.get_effective_tabstop(bufnr)
runtime/lua/vim/lsp/util.lua|1917 col 17| tabSize = M.get_effective_tabstop();
test/functional/plugin/lsp_spec.lua|2376 col 22| describe('lsp.util.get_effective_tabstop', function()
test/functional/plugin/lsp_spec.lua|2383 col 49| eq(tabsize, exec_lua('return vim.lsp.util.get_effective_tabstop()'))
```

**Comment by tpope** [comment]
Personally I'd shoot down `get_effective_indentsize`; adding a third term is just inviting confusion. I'd stick to `get_effective_tabsize`,or if I wanted to avoid the "tab" baggage, go back to idiomatic Vim: `get_effective_shiftwidth`.

This is a documented function. I take it you're fine with the backwards compatibility break? I doubt it has many consumers. Your call on that and the name; let me know and I can manage the search and replace.

**Comment by tpope** [comment]
Regardless the function documentation will need tweaking too, I'll handle that at the same time.

**Comment by mjlbach** [comment]
> Personally I'd shoot down get_effective_indentsize; adding a third term is just inviting confusion. I'd stick to get_effective_tabsize,or if I wanted to avoid the "tab" baggage, go back to idiomatic Vim: get_effective_shiftwidth.

Completely agree, I should have said shiftwidth :) 

> This is a documented function. I take it you're fine with the backwards compatibility break? 

What we typically do for this is use `notify_once`, and alias the old function to the new function. I did a search and it seems like an unexpectedly large number of users are calling it in their configs (mostly for tailwind). I'm also totally ok with doing that in a separate PR (I can also take care of that) and just merging this PR as is for now!

**Review by mjlbach — APPROVED** [review]

**Comment by tpope** [comment]
Sounds good. I've amended a low effort documentation tweak.

**Comment by mjlbach** [comment]
Great! Thanks for updating that. I will merge once the tests are green.

File: runtime/doc/lsp.txt
Comment (zeertzjq) [inline runtime/doc/lsp.txt:1461]: Hmm, this is generated from the doxygen comments.

Comment (mjlbach) [reply runtime/doc/lsp.txt:1461]: @tpope line 1901 and co need to get updated, then if you re-run the document generation pipeline it should overwrite the manual changes.

Comment (tpope) [reply runtime/doc/lsp.txt:1461]: Good catch, I'll fix that.

**Pull request merged**
